src/packagedcode/swift.py

```diff
@@ -47,12 +47,75 @@ def logger_debug(*args):
         return logger.debug(" ".join(isinstance(a, str) and a or repr(a) for a in args))
 
 
+class SwiftShowDependenciesDepLockHandler(models.DatafileHandler):
+    datasource_id = "swift_package_show_dependencies"
+    path_patterns = ("*/swift-show-dependencies.deplock",)
+    default_package_type = "swift"
+    default_primary_language = "Swift"
+    description = "Swift dependency graph created by DepLock"
+    documentation_url = "https://forums.swift.org/t/swiftpm-show-dependencies-without-fetching-dependencies/51154"
+
+    @classmethod
+    def _parse(cls, swift_dependency_relation, package_only=False):
+
+        if TRACE:
+            logger_debug(
+                f"SwiftShowDependenciesDepLockHandler: deplock: package: {swift_dependency_relation}"
+            )
+
+        dependencies = get_flatten_dependencies(
+            swift_dependency_relation.get("dependencies")
+        )
+
+        package_data = dict(
+            datasource_id=cls.datasource_id,
+            type=cls.default_package_type,
+            primary_language=cls.default_primary_language,
+            namespace=None,
+            name=swift_dependency_relation.get("name"),
+            dependencies=dependencies,
+        )
+
+        return models.PackageData.from_data(package_data, package_only)
+
+    @classmethod
+    def parse(cls, location, package_only=False):
+        with io.open(location, encoding="utf-8") as loc:
+            swift_dependency_relation = json.load(loc)
+
+        yield cls._parse(swift_dependency_relation, package_only)
+
+    @classmethod
+    def assemble(
+        cls, package_data, resource, codebase, package_adder=models.add_to_package
+    ):
+        siblings = resource.siblings(codebase)
+        swift_manifest_resource = [
+            r
+            for r in siblings
+            if r.name in ("Package.swift.json", "Package.swift.deplock")
+        ]
+
+        # Skip the assembly if the Swift manifest is present.
```

Comment:
This is not the correct way to do assembly. See https://github.com/nexB/scancode-toolkit/blob/develop/src/packagedcode/npm.py#L70 or other parsers on creating a base handler class with assembly and all other manifests inheriting from there.
From there look for the main manifest in sibling files from the location, and parse and yield from there accordingly. By this way we assemble from the first package that gets picked up, instead of walking the codebase further for more manifests.
Just a nit, assembly is working correctly otherwise.

Reply:
Tracking this here https://github.com/nexB/scancode-toolkit/issues/3837